File: OptiSched-Web/src/types/Institution.ts
export type SubscriptionStatus = "TRIAL" | "ACTIVE" | "CANCELED" | "UNPAID";

export type InstitutionType = "UNIVERSITY" | "SCHOOL";

export type Institution = {
  id: number;
  name: string;
  cnpj: string;
  subscriptionStatus: SubscriptionStatus;
  expiresAt: string | null;
  type: InstitutionType;
};

export type InstitutionInput = {
  name: string;
  cnpj: string;
  subscriptionStatus: SubscriptionStatus;
  expiresAt: string | null;
  type: InstitutionType;
};

File: OptiSched-API/src/main/java/com/vinibarros/optisched/entity/Institution.java
package com.vinibarros.optisched.entity;

import com.vinibarros.optisched.enums.InstitutionType;
import com.vinibarros.optisched.enums.SubscriptionStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "institution")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Institution extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true, length = 100)
    private String slug;

    @Column(length = 14, unique = true)
    private String cnpj;

    @Column(name = "subscription_id")
    private String subscriptionId;

    @Enumerated(EnumType.STRING)
    @Column(name = "subscription_status", nullable = false, length = 20)
    private SubscriptionStatus subscriptionStatus = SubscriptionStatus.TRIAL;

    @Column(name = "expires_at")
    private LocalDateTime expiresAt;

    @Enumerated(EnumType.STRING)
    @Column(name = "type", nullable = false, length = 20)
    private InstitutionType type = InstitutionType.UNIVERSITY;

    @Column(name = "is_demo", nullable = false)
    private boolean demo = false;
}

